src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -194,4 +196,19 @@ public ResponseEntity<ZonedDateTime> getIrisAcceptedForStudent() {
         User user = userRepository.getUser();
         return ResponseEntity.ok().body(user.getIrisAcceptedTimestamp());
     }
+
+    /**
+     * PUT users/sshpublickey : sets the ssh public key
+     *
+     * @return the ResponseEntity with status 200 (OK), with status 404 (Not Found), or with status 400 (Bad Request)
+     */
+    @PutMapping("users/sshpublickey")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Void> addSshPublicKey(String sshPublicKey) throws NoSuchAlgorithmException {
+        User user = userRepository.getUser();
+        var sshPublicKeyHash = HashUtils.hashString(sshPublicKey);
```

Comment:
We should not hash the public key

Reply:
Hashing the key is needed for fast lookup in the database. We now store the hashed value for fast lookup with a database index and the full key to display it in the client. This should be a good compromise